### ISSUE TITLE
Bug accessing osquery VM on Linux using OSX

### ISSUE THREAD

**Issue by dm-17** [facebook/osquery#1559]
When connecting to a linux VM (ubuntu 14.04.1 and ossuary 1.5.3 - provided during Brucon), from an OS X (v 10.11) using SSH, the following bug occurs:

$ osqueryi
terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
Aborted (core dumped)

My troubleshooting analysis lead me to the following workaround:
remove the env variable LC_CTYPE with the default value of "LC_CTYPE=UTF-8":
unset LC_CTYPE

$ osqueryi
osquery - being built, with love, at Facebook
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Using a virtual database. Need help, type '.help'

**Comment by theopolis** [comment]
Very interesting bug! Thanks for the debugging/analysis, a missing environment variable should NOT cause an unhandled exception, we'll get this fixed ASAP!

**Comment by theopolis** [comment]
Here's how I can reproduce:
```
~/git/github/osquery (branch:fix_tests_3) » LC_CTYPE=UTF-81111            
~/git/github/osquery (branch:fix_tests_3) » lldb ./build/debug_centos7/osquery/osqueryi
Current executable set to './build/debug_centos7/osquery/osqueryi' (x86_64).
(lldb) run
Process 3603 launched: './build/debug_centos7/osquery/osqueryi' (x86_64)
(lldb) terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
Process 3603 stopped
* thread #1: tid = 3603, 0x00007f66e09ce5d7 libc.so.6`__GI_raise + 55, name = 'osqueryi', stop reason = signal SIGABRT
    frame #0: 0x00007f66e09ce5d7 libc.so.6`__GI_raise + 55
libc.so.6`__GI_raise + 55:
-> 0x7f66e09ce5d7:  cmpq   $-0x1000, %rax
   0x7f66e09ce5dd:  ja     0x7f66e09ce5fd            ; __GI_raise + 93
   0x7f66e09ce5df:  rep    
   0x7f66e09ce5e0:  ret    

error: No auto repeat.
(lldb) bt
* thread #1: tid = 3603, 0x00007f66e09ce5d7 libc.so.6`__GI_raise + 55, name = 'osqueryi', stop reason = signal SIGABRT
  * frame #0: 0x00007f66e09ce5d7 libc.so.6`__GI_raise + 55
    frame #1: 0x00007f66e09cfcc8 libc.so.6`__GI_abort + 328
    frame #2: 0x00007f66e12d29b5 libstdc++.so.6`__gnu_cxx::__verbose_terminate_handler() + 357
    frame #3: 0x00007f66e12d0926 libstdc++.so.6`??? + 6
    frame #4: 0x00007f66e12d0953 libstdc++.so.6`std::terminate() + 19
    frame #5: 0x00007f66e12d0b73 libstdc++.so.6`__cxa_throw + 115
    frame #6: 0x00007f66e1325a87 libstdc++.so.6`std::__throw_runtime_error(char const*) + 119
    frame #7: 0x00007f66e12eea14 libstdc++.so.6`std::locale::facet::_S_create_c_locale(__locale_struct*&, char const*, __locale_struct*) + 36
    frame #8: 0x00007f66e12e38dc libstdc++.so.6`std::locale::_Impl::_Impl(char const*, unsigned long) + 92
    frame #9: 0x00007f66e12e4a4b libstdc++.so.6`std::locale::locale(char const*) + 1835
    frame #10: 0x0000000000dc7b8c osqueryi`boost::filesystem::path::codecvt() + 44
    frame #11: 0x0000000000ae45d5 osqueryi`path<char *>(this=0x00007fffaad35d68, source=0x00007fffaad35d60, (null)=0x0000000000000000) + 69 at path.hpp:139
    frame #12: 0x0000000000ae282c osqueryi`osquery::osqueryHomeDirectory() + 252 at filesystem.cpp:353
    frame #13: 0x0000000000a03fd0 osqueryi`Initializer(this=0x00007fffaad36098, argc=0x00007fffaad360c0, argv=0x00007fffaad360b8, tool=OSQUERY_TOOL_SHELL) + 1072 at init.cpp:239
    frame #14: 0x00000000008c3a23 osqueryi`main(argc=1, argv=0x00007fffaad361b8) + 67 at shell.cpp:21
    frame #15: 0x00007f66e09baaf5 libc.so.6`__libc_start_main + 245
```

**Comment by theopolis** [comment]
To be clear, this isn't an unbreak-now or a vulnerability as it requires control of the running user's environment. Much worse things are possible if that is the case. This is just super annoying since many VMs/containers will may not support the provided environment/language.

**Comment by theopolis** [comment]
This issue is fixed in boost 1.56.0: https://svn.boost.org/trac/boost/ticket/4688. 

The versions of boost used are as follows:
CentOS 6: 1.55 (local build)
CentOS 7: 1.53 (EL yum)
Ubuntu 12: 1.55 (local build)
Ubuntu 14: 1.55/1.54 (apt source)

We can try to set the local explicitly during initialize after attempt to induce a boost crash.
